lesson34-arrays/index.js

```diff
@@ -0,0 +1,162 @@
+function markProductPurchased(arrayOfProducts, nameOfProduct) {
+  return arrayOfProducts.map((product) => {
+    return product.productName === nameOfProduct ? { ...product, bought: true } : { ...product };
+  });
+}
+console.log(markProductPurchased(shopingList, "cheese"));
+console.log(markProductPurchased(shopingList, "orange"));
+
+//* NORMA
+/*
+1. Видалення продукту зі списку (видалення повинно проводитися шляхом створення нового масиву, 
+  в якому продукт, що ми шукаємо, буде відсутнім)*/
+
+const deleteProduct = (shopingList, nameOfproduct) => {
+  const newListOfProducts = [];
+  shopingList.forEach((product) => {
+    if (product.productName !== nameOfproduct) {
+      newListOfProducts.push(product);
+    }
+  });
+
+  return newListOfProducts;
+};
+
+console.log(deleteProduct(shopingList, "cheese"));
+
+/*2. Додавання покупки в список. Враховуй, що при додаванні покупки з уже існуючим в списку продуктом, 
+необхідно збільшувати кількість в існуючій покупці, а не додавати нову. При цьому також повинна 
+змінитися сума, наприклад, якщо ціна за одиницю 12, а кількості товарів стало 2, то сума буде 24
+*/
+function addProductToList(shopingList, nameOfproduct, qtyOfProduct, pricePerItem) {
+  const index = shopingList.findIndex((product) => product.productName === nameOfproduct);
+  if (index !== -1) {
+    shopingList[index].qty += qtyOfProduct;
+    shopingList[index].pricePerItem = pricePerItem;
+    shopingList[index].amount = shopingList[index].qty * shopingList[index].pricePerItem;
+  } else {
+    shopingList.push({
+      productName: nameOfproduct,
+      qty: qtyOfProduct,
+      bought: false,
+      pricePerItem: pricePerItem,
+      amount: qtyOfProduct * pricePerItem,
+    });
+  }
+
+  return shopingList;
+}
+console.log(addProductToList(shopingList, "bread", 2, 27));
+console.log(addProductToList(shopingList, "banana", 3, 60));
+console.log(addProductToList(shopingList, "milk", 3, 37));
+
+// * MAX
+/*
+/*1. Підрахунок суми всіх продуктів (враховуючи кількість кожного) в списку.*/
+
+function calcSummOfAllProducts(shopingList) {
+  const listOfAmountAllProducts = shopingList.map((product) => product.amount);
+  const sumOfAllProducts = listOfAmountAllProducts.reduce((acc, cur) => acc + cur);
+  return `Sum of all products is ${sumOfAllProducts}`;
+}
+
+console.log(calcSummOfAllProducts(shopingList));
+
+/*2.Підрахунок суми всіх (не) придбаних продуктів.*/
+function calcSumBughtProducrs(shopingList) {
+  const sumBoughtProducts = shopingList
+    .filter((product) => product.bought === true)
+    .map((product) => product.amount)
+    .reduce((acc, cur) => acc + cur);
+  console.log(`Sum of all bought products is ${sumBoughtProducts}`);
+}
+calcSumBughtProducrs(shopingList);
+
+/*3. Показання продуктів в залежності від суми, (від більшого до меншого / від меншого до більшого, 
+в залежності від параметра функції, який вона приймає)
+*/
+
+function sortBySum(array, sortMethod) {
+  if (sortMethod === 1) {
```

Comment:
what is 1 and 2 ??
how for instance other developered shoudl know what to pass 
it is better to pass 'ascending ' and 'descending '